File: src/main/java/com/pubnub/api/models/consumer/objects_api/user/PNUser.java
package com.pubnub.api.models.consumer.objects_api.user;

import com.pubnub.api.models.consumer.objects_api.PNObject;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;

@Getter
@Setter
@Accessors(chain = true)
@EqualsAndHashCode(callSuper = true)
public class PNUser extends PNObject {

    private String name;
    private String externalId;
    private String profileUrl;
    private String email;

    public PNUser(String id, String name) {
        super(id);
        this.name = name;
    }

    @Override
    public PNUser setCustom(Object custom) {
        super.setCustom(custom);
        return this;
    }
}

File: src/main/kotlin/com/pubnub/api/models/consumer/objects_api/user/PNUser.kt
package com.pubnub.api.models.consumer.objects_api.user

import com.pubnub.api.models.consumer.objects_api.PNObject

class PNUser(id: String, var name: String) : PNObject(id) {

    var externalId: String? = null
    var profileUrl: String? = null
    var email: String? = null

    override fun toString(): String {
        return "PNUser(" +
                "name='$name', " +
                "externalId=$externalId, " +
                "profileUrl=$profileUrl, " +
                "email=$email" +
                ")" +
                " ${super.toString()}"
    }

}